contrib/natsio/src/main/java/com/google/cloud/dataflow/contrib/natsio/NatsIO.java

```diff
@@ -0,0 +1,395 @@
+ * } </pre>
+ * 
+ * <p>or
+ * <pre> {@code
+ * 		options.setStreaming(true);
+ * 
+ * 		Pipeline p = Pipeline.create(options);
+ *		p
+ *			.apply(NatsIO.Read.from(subject, props)) // Streaming
+ *			.apply(ParDo.of(new NatsIOBench.SimpleFunc()));
+ * } </pre>
+ * 
+ */
+public class NatsIO {
+	private static final Logger LOG = LoggerFactory.getLogger(NatsIO.class);
+	
+	private static final Coder<String> DEFAULT_CODER = StringUtf8Coder.of();
+	private static Read.NatsSource.CommitMark DEFAULT_MARK = new Read.NatsSource.CommitMark();
+	private static final byte[] DEFAULT_ID = new byte[1];
+	
+	public static class Read {
+		/**
+		 * Create a unbounded NatsSource instance. 
+		 * @param subject to subscribe and receive updates
+		 * @param props contains NATS related properties. "servers" property is mandatory to access a remote NATS server.
+		 * @return a unbounded NatsSource instance
+		 */
+		public static com.google.cloud.dataflow.sdk.io.Read.Unbounded<KV<String,String>> from(String subject, Properties props) {
+			return com.google.cloud.dataflow.sdk.io.Read.from(new NatsSource(subject, props));
+		}
+
+		/**
+		 * Create a NatsSource instance which is bounded to a number of updates to receive. 
+		 * @param subject to subscribe and received updates
+		 * @param numRecords bounds a number of records to receive from NATS server
+		 * @param props contains NATS related properties. "servers" property is mandatory to access a remote NATS server.
+		 * @return a bounded NatsSource instance
+		 */
+		public static PTransform<PInput, PCollection<KV<String,String>>> withMaxNumRecords(String subject, long numRecords, Properties props) {
+			return from(subject, props).withMaxNumRecords(numRecords);
+		}		
+
+		/**
+		 * Create a NatsSource instance which is bounded to total reading time to receive updates.
+		 * @param subject to subscribe and received updates
+		 * @param readTime bounds total reading time to receive updates from NATS server.
+		 * @param props contains NATS related properties. "servers" property is mandatory to access a remote NATS server.
+		 * @return a bounded NatsSource instance
+		 */
+		public static PTransform<PInput, PCollection<KV<String,String>>> withMaxReadtime(String subject, long readTime, Properties props) {
+			return from(subject, props).withMaxReadTime(Duration.standardSeconds(readTime));
+		}
+		
+		public static class NatsSource extends UnboundedSource<KV<String,String>, UnboundedSource.CheckpointMark> {
+			private static final long serialVersionUID = 1L;
+			
+			private Properties props = null;
+			private Connection conn = null;
+			private String subject;
+			
+			public NatsSource(String subject, Properties props) {
+				if (subject == null) {
+					throw new NullPointerException("subject");
+				}
+				if (props == null) {
+					throw new NullPointerException("props");
+				}
+				if (props.getProperty("servers") == null) {
+					throw new NullPointerException("servers");
```

Comment:
I would probably change this one to a `Preconditions.checkArgument` instead, since it's not a direct parameters.

```
import static com.google.common.base.Preconditions.checkArgument;

...

checkArgument(props.getProperty("servers") != null, "Expected property \"servers\" in the provided properties");
```